kafka-plugins/src/main/java/co/cask/hydrator/plugin/realtime/KafkaProducer.java

```diff
@@ -0,0 +1,278 @@
+            printer = new CSVPrinter(writer, csvFileFormat);
+            break;
+          
+          case "mysql":
+            csvFileFormat = CSVFormat.Predefined.MySQL.getFormat();
+            printer = new CSVPrinter(writer, csvFileFormat);
+            break;
+          
+          case "tdf":
+            csvFileFormat = CSVFormat.Predefined.TDF.getFormat();
+            printer = new CSVPrinter(writer, csvFileFormat);
+            break;
+          
+          case "rfc4180":
+            csvFileFormat = CSVFormat.Predefined.TDF.getFormat();
+            printer = new CSVPrinter(writer, csvFileFormat);
+            break;
+        }
+        
+        if (printer != null) {
+          printer.printRecord(objs);
+          body = writer.toString();
+        }
+      }
+      
+      // Message key.
+      String key = "no_key";
+      if (producerConfig.key != null) {
+        key = object.get(producerConfig.key);
+      }
+      
+      // Extract the partition key from the record. If the partition key is 
+      // Integer then we use it as-is else
+      int partitionKey = 0;
+      if (producerConfig.partitionField != null) {
+        if(object.get(producerConfig.partitionField) != null) {
+          partitionKey = object.get(producerConfig.partitionField).hashCode();
+        }
+      }
+
+      // Write to all the configured topics
+      for (String topic : topics) {
+        partitionKey = partitionKey % producer.partitionsFor(topic).size();
+        if (isAsync) {
+          producer.send(new ProducerRecord<String, String>(topic, partitionKey, key, body), new Callback() {
+            @Override
+            public void onCompletion(RecordMetadata meta, Exception e) {
+              if (meta != null) {
+                context.getMetrics().count("kafka.async.success", 1);
+              }
+              
+              if (e != null) {
+                context.getMetrics().count("kafka.async.error", 1);  
+              }
+            }
+          });
+        } else {
+          // Waits infinitely to push the message through. 
+          producer.send(new ProducerRecord<String, String>(topic, partitionKey, key, body)).get();
+        }
+        context.getMetrics().count("kafka.producer.count", 1);
+      }
+    }
+    return count;
+  }
+  
+  @Override
+  public void destroy() {
+    try {
+      super.destroy();
+    } finally {
+      if (producer != null) {
+        producer.close();
+      }
+    }
+  }
+
+  /**
+   * Kafka Producer Configuration.
+   */
+  public static class Config extends PluginConfig {
+    
+    @Name("brokers")
+    @Description("Specifies the connection string where Producer can find one or more brokers to " +
+      "determine the leader for each topic")
+    private String brokers;
+    
+    @Name("isasync")
+    @Description("Specifies whether an acknowledgment is required from broker that message was received. " +
+      "Default is FALSE")
+    private String isAsync;
```

Comment:
Why not use boolean type? Also, better call it "async" instead of "isAsync" (java bean naming convention).